pkg/oauth/server/osinserver/registrystorage/storage.go

```diff
@@ -84,6 +84,13 @@ func (w *clientWrapper) GetUserData() interface{} {
 	return w.client
 }
 
+func (w *clientWrapper) GetExpiration(defaultExpiration int32) int32 {
+	if w.client.AccessTokenMaxAgeSeconds != nil {
```

Comment:
they are admin-controlled today, especially since grant method can be set to auto accept per-client